metrics/firebase/functions/test/main_test.dart

```diff
@@ -0,0 +1,250 @@
+
+    final expectedBuildDayData = {
+      'projectId': projectId,
+      buildDayStatusFieldName: 1,
+      'totalDuration': durationInMilliseconds,
+      'day': Timestamp.fromDateTime(utcDate),
+    };
+
+    final exception = Exception('test');
+    final expectedCreatedTask = {
+      'code': 'build_days_created',
+      'data': build,
+      'context': exception.toString(),
+      'createdAt': Timestamp.fromDateTime(utcDate),
+    };
+
+    final _firestoreMock = FirestoreMock();
+    final _collectionReferenceMock = CollectionReferenceMock();
+    final _documentReferenceMock = DocumentReferenceMock();
+    final _documentSnapshotMock = DocumentSnapshotMock();
+    final _eventContextMock = EventContextMock();
+
+    tearDown(() {
+      reset(_documentSnapshotMock);
+      reset(_eventContextMock);
+      reset(_firestoreMock);
+      reset(_collectionReferenceMock);
+      reset(_documentReferenceMock);
+    });
+
+    PostExpectation<Firestore> whenFirestore() {
+      return when(_documentSnapshotMock.firestore);
+    }
+
+    PostExpectation<CollectionReference> whenCollection(String withName) {
+      whenFirestore().thenReturn(_firestoreMock);
+
+      return when(_firestoreMock.collection(withName));
+    }
+
+    PostExpectation<DocumentReference> whenDocument({
+      String withCollectionName,
+    }) {
+      whenCollection(withCollectionName).thenReturn(_collectionReferenceMock);
+
+      return when(_collectionReferenceMock.document(any));
+    }
+
+    PostExpectation<Future<void>> whenDocumentData({
+      String withCollectionName,
+    }) {
+      whenDocument(
+        withCollectionName: withCollectionName,
+      ).thenReturn(_documentReferenceMock);
+
+      return when(_documentReferenceMock.setData(any, any));
+    }
+
+    PostExpectation whenSnapshotData() {
+      return when(_documentSnapshotMock.data);
+    }
+
+    test(
+      "sets a build day document with the zero duration if the build document snapshot's duration is a null",
+      () async {
+        whenDocumentData(withCollectionName: buildDaysCollectionName)
+            .thenAnswer((_) => Future.value());
+
+        final Map<String, dynamic> buildWithoutDuration = Map.from(build);
+        buildWithoutDuration.remove('duration');
+
+        whenSnapshotData()
+            .thenReturn(DocumentData.fromMap(buildWithoutDuration));
+
+        await onBuildAddedHandler(_documentSnapshotMock, _eventContextMock);
+
+        final documentDataMatcher = predicate<DocumentData>((data) {
+          return data.getNestedData('totalDuration').getInt('operand') == 0;
+        });
+
+        final optionMatcher = predicate<SetOptions>((option) => option.merge);
+
+        verify(
+          _documentReferenceMock.setData(
+            argThat(documentDataMatcher),
+            argThat(optionMatcher),
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "uses a composite document id for the build days collection",
```

Comment:
Should we test that this method trims the time part of the `startedAt` DateTime and converts it to UTC to create a document identifier?